### PR TITLE
[GLUTEN-2498] Remove `FileScan` in `HiveTableScanExecTransformer`

### PR DESCRIPTION
**Pull request by baibaichen**
## What changes were proposed in this pull request?

Currently, We  use `CatalogFileIndex` to list files, it eventually use `InMemoryFileIndex` which could create spark job. It is much slower than vanilla spark's implemation which uses `FileInputFormat.getSplits()`

This PR simulates  `FileInputFormat.getSplits()`,  but only use one thread. We can support mutiple threads latter.

(Fixes: \#2498)

## How was this patch tested?
UT

### CODE REVIEW

File: gluten-core/src/main/scala/org/apache/spark/sql/execution/datasources/GlutenTextBasedScanWrapper.scala [hunk 1]
```diff
@@ -1,92 +0,0 @@
...
-
-  def withFilters(partitionFilters: Seq[Expression], dataFilters: Seq[Expression]): FileScan = {
-    scan match {
-      case t: TextScan => t.copy(partitionFilters = partitionFilters, dataFilters = dataFilters)
-      case j: JsonScan => j.copy(partitionFilters = partitionFilters, dataFilters = dataFilters)
-      case _ => scan
-    }
-  }
-
-  def compressionSplittable(path: Path): Boolean = {
```
Comment (KevinyhZou) [inline gluten-core/src/main/scala/org/apache/spark/sql/execution/datasources/GlutenTextBasedScanWrapper.scala:70]: Hello, Perhaps we should consider the compression text file not support split currently, like bzip2, lz4 which not support split by clickhouse. So it best to set unsplittable when the text file is compressed

Comment (baibaichen) [reply gluten-core/src/main/scala/org/apache/spark/sql/execution/datasources/GlutenTextBasedScanWrapper.scala:70]: Add test 
image

and using `BackendsApiManager` to decide whether split or not
image

**Comment by KevinyhZou** [comment]
LGTM

**Review by zzcclp — APPROVED** [review]
LGTM

**Pull request merged**